x-pack/plugins/infra/server/lib/alerting/log_threshold/log_threshold_chart_preview.ts

```diff
@@ -97,10 +98,20 @@ const addHistogramAggregationToQuery = (
   };
 
   if (isGrouped) {
-    query.body.aggregations.groups.aggregations.filtered_results = {
-      ...query.body.aggregations.groups.aggregations.filtered_results,
-      aggregations: histogramAggregation,
-    };
+    const isOptimizedQuery = (groupedQuery: any) =>
+      !groupedQuery.body.aggregations.groups.aggregations?.filtered_results ? true : false;
+
+    if (isOptimizedQuery(query)) {
```

Comment:
I feel a bit bad talking about this small section so much :see_no_evil: so please ignore if inappropriate, but there might be a quick way to avoid the extra `any` while preserving the self-documenting character of the condition. How about

```suggestion
    const isOptimized = query.body.aggregations.groups.aggregations?.filtered_results != null;

    if (isOptimized) {
```

?